db/migrate/20240312122903_make_stripe_payment_intent_generic.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class MakeStripePaymentIntentGeneric < ActiveRecord::Migration[7.1]
+  def change
+    change_table :stripe_payment_intents do |t|
+      t.rename :user_id, :initiated_by_id # Make the purpose of the user_id column more clear
+      t.rename :confirmed_by_id, :confirmation_source_id
+      t.rename :confirmed_by_type, :confirmation_source_type
+      t.rename :canceled_by_id, :cancellation_source_id
+      t.rename :canceled_by_type, :cancellation_source_type
+
+      t.string :wca_status
+      t.string :payment_record_type
+      t.integer :payment_record_id
+    end
+
+    rename_table :stripe_payment_intents, :payment_intents
+    rename_column :stripe_records, :status, :stripe_status
+    rename_column :stripe_records, :api_type, :record_type
+    rename_column :paypal_records, :status, :paypal_status
+
+    reversible do |direction|
+      direction.up do
+        PaymentIntent.update_all(payment_record_type: 'StripeRecord')
+
+        PaymentIntent.find_each do |intent|
+          intent.assign_attributes(payment_record_id: intent.stripe_record_id)
+          intent.set_wca_status
+          intent.save
+        end
+      end
+
+      direction.down do
+        PaymentIntent.find_each do |intent|
+          intent.assign_attributes(stripe_record_id: intent.payment_record_id)
+          intent.save(validate: false)
```

Comment:
This is literally exactly what `intent.update_attribute(stripe_record_id: intent.payment_record_id)` was designed to do. (It implicitly saves and skips validations)